src/appkit.cs

```diff
@@ -10268,7 +10268,7 @@ partial interface NSImageRep : NSCoding, NSCopying {
 		bool DrawInRect (CGRect rect);
 
 		[Export ("drawInRect:fromRect:operation:fraction:respectFlipped:hints:")]
-		bool DrawInRect (CGRect dstSpacePortionRect, CGRect srcSpacePortionRect, NSCompositingOperation op, nfloat requestedAlpha, bool respectContextIsFlipped, NSDictionary hints);
+		bool DrawInRect (CGRect dstSpacePortionRect, CGRect srcSpacePortionRect, NSCompositingOperation op, nfloat requestedAlpha, bool respectContextIsFlipped, [NullAllowed] NSDictionary hints);
```

Comment:
ignore file for `xtro` should be updated as well:
https://github.com/xamarin/xamarin-macios/blob/b540e1f2b9940af69ac5dbf813be3602bb6e3c3a/tests/xtro-sharpie/macOS-AppKit.ignore#L1272

There also seems to be a similar method in `NSStringDrawing_NSString`:
https://github.com/xamarin/xamarin-macios/blob/b540e1f2b9940af69ac5dbf813be3602bb6e3c3a/tests/xtro-sharpie/macOS-AppKit.ignore#L1300

Reply:
Thanks for the (super prompt) review! I'll update those as well.

Reply:
@Therzok Hm, the [NSString.DrawInRect](https://developer.apple.com/documentation/foundation/nsstring/1529855-drawinrect) docs don't say anything about being optional, so I just updated the ignore for this particular API.

Reply:
I ran `open -h NSStringDrawing.h` and looked at `drawInRect:`.

This is the native definition for NSStringDrawing protocol implementation, the attributes are indeed nullable. I've noticed that most of the time the web documentation is not complete, where-as the headers do have all the annotations.

```objc
@interface NSString(NSStringDrawing)
- (NSSize)sizeWithAttributes:(nullable NSDictionary<NSAttributedStringKey, id> *)attrs API_AVAILABLE(macos(10.0), ios(7.0), watchos(2.0), tvos(9.0));
- (void)drawAtPoint:(NSPoint)point withAttributes:(nullable NSDictionary<NSAttributedStringKey, id> *)attrs API_AVAILABLE(macos(10.0), ios(7.0), watchos(2.0), tvos(9.0));
- (void)drawInRect:(NSRect)rect withAttributes:(nullable NSDictionary<NSAttributedStringKey, id> *)attrs API_AVAILABLE(macos(10.0), ios(7.0), watchos(2.0), tvos(9.0));
@end
```

Reply:
Ah, ok.  I can update those as well then.  Though it looks like quite a large rabbit hole I'm getting into here 😅

Reply:
Ok, updates are done!  let me know if you'd like any other changes.

Reply:
xtro will tell you since it looks at the data present in the headers, so if llvm knows about it, xtro will.

@cwensley  I'll keep an eye on the CI for you